File: 2340-total-appeal-of-a-string/total-appeal-of-a-string.java
class Solution {
    public long appealSum(String s) {
        long nums = 0;
        long total = 0;
        long[] prev = new long[26];
        Arrays.fill(prev, -1);

        for (int i = 0; i < s.length(); i++) {
            nums = nums + (i - 0 - prev[s.charAt(i) - 'a']);
            prev[s.charAt(i) - 'a'] = i;
            total += nums;
        }
        return total;
    }
}

File: 2340-total-appeal-of-a-string/total-appeal-of-a-string.py
class Solution:
    def appealSum(self, s: str) -> int:
        #c, b, c, d, e, c
        last_pos = [-1] * 26
        dp = 0
        total = 0
        for i in range(len(s)):
            char_idx = ord(s[i]) - ord('a')
            if last_pos[char_idx] == -1:
                dp = dp + 1 + i
            else:
                dp = dp + (i - last_pos[char_idx])
            last_pos[ord(s[i]) - ord('a')] = i
            total += dp
        return total